apps/cowswap-frontend/src/modules/hooksStore/hooks/useBalancesDiff.ts

```diff
@@ -0,0 +1,124 @@
+    if (!data || !preHooks.length) return EMPTY_BALANCE_DIFF
+
+    const lastPreHook = preHooks[preHooks.length - 1]
+    return data[lastPreHook?.uuid]?.cumulativeBalancesDiff || EMPTY_BALANCE_DIFF
+  }, [data, preHooks])
+}
+
+// Returns all the ERC20 Balance Diff of the current hook to be passed to the iframe context
+export function useHookBalancesDiff(isPreHook: boolean, hookToEditUid?: string): BalancesDiff {
+  const { account } = useWalletInfo()
+  const { data } = useTenderlyBundleSimulation()
+  const orderParams = useOrderParams()
+  const { preHooks, postHooks } = useHooks()
+  const preHookBalanceDiff = usePreHookBalanceDiff()
+
+  const orderMockBalanceDiff = useMemo(() => {
+    if (!account) return EMPTY_BALANCE_DIFF
+    const balanceDiff: Record<string, string> = {}
+
+    if (orderParams?.buyAmount && orderParams.buyTokenAddress && account)
+      balanceDiff[orderParams.buyTokenAddress.toLowerCase()] = orderParams.buyAmount
+
+    if (orderParams?.sellAmount && orderParams.sellTokenAddress && account)
+      balanceDiff[orderParams.sellTokenAddress.toLowerCase()] = `-${orderParams.sellAmount}`
+
+    return { account: balanceDiff }
+  }, [orderParams, account])
+
+  const firstPostHookBalanceDiff = useMemo(() => {
+    return mergeBalanceDiffs(preHookBalanceDiff, orderMockBalanceDiff)
+  }, [preHookBalanceDiff, orderMockBalanceDiff])
+
+  const postHookBalanceDiff = useMemo(() => {
+    // is adding the first post hook or simulation not available
+    if (!data || !postHooks) return firstPostHookBalanceDiff
+
+    const lastPostHook = postHooks[postHooks.length - 1]
+    return data[lastPostHook?.uuid]?.cumulativeBalancesDiff || firstPostHookBalanceDiff
+  }, [data, postHooks, orderMockBalanceDiff, preHookBalanceDiff])
+
+  const hookToEditBalanceDiff = useMemo(() => {
+    if (!data || !hookToEditUid) return EMPTY_BALANCE_DIFF
+
+    const otherHooks = isPreHook ? preHooks : postHooks
+
+    const hookToEditIndex = otherHooks.findIndex((hook) => hook.uuid === hookToEditUid)
+
+    // is editing first preHook -> return empty state
+    if (!hookToEditIndex && isPreHook) return EMPTY_BALANCE_DIFF
+
+    // is editing first postHook -> return
+    if (!hookToEditIndex && !isPreHook) return firstPostHookBalanceDiff
+
+    // is editing a non first hook, return the latest available hook state
+    const previousHookIndex = hookToEditIndex - 1
+
+    return data[otherHooks[previousHookIndex]?.uuid]?.cumulativeBalancesDiff || EMPTY_BALANCE_DIFF
+  }, [data, hookToEditUid, isPreHook, preHooks, postHooks, firstPostHookBalanceDiff])
+
+  return useMemo(() => {
+    if (hookToEditUid) return hookToEditBalanceDiff
+    if (isPreHook) return preHookBalanceDiff
+    return postHookBalanceDiff
+  }, [data, orderParams, preHooks, postHooks])
+}
+
+const addBigNumberStrings = (a: string, b: string): string => {
+  const bigA = BigNumber.from(a)
+  const bigB = BigNumber.from(b)
+  return bigA.add(bigB).toString()
+}
```

Comment:
this looks like a utility function that should be somewhere else?

Reply:
I think that this utility was so simple and just used once that we don't need it actually 😅